mozjs/js/rust/src/magic.rs

```diff
@@ -1,33 +1,73 @@
 //! Common traits and types for Magic DOM structs.
 
-/// the proper name is the name of the struct generated by the MagicDom macro
+/// The proper name is the name of the struct generated by the MagicDom macro.
+/// The class_name is the name of the JSClass generated by the MagicDom macro.
+/// The ps_arr_name is the name of the JSPropertySpec array generated by the MagicDom macro.
+/// The constructor name is the name of the constructor for the JSClass generated by the
+/// MagicDom macro.
+/// The mod_name is the name of the private module.
+/// For example, suppose the name of the dom struct is DOMPoint($name) and the upper
+/// case version is DOMPOINT($NAME). We will put the following in the magic_dom!:
+/// magic_dom! {
+///     DOMPoint,
+///     DOMPOINT_CLASS, // the format for the class name would be $NAME + _CLASS
+///     DOMPOINT_PS_ARR, // the format for the JSPropertySpec array would be $NAME + _PS_ARR
+///     DOMPoint_constructor, // the format for the constructor would be $name + _constructor
+///     // the format for the private module name would be magic_dom_spec_ + $name
+///     magic_dom_spec_DOMPoint,
+///     // the spec name format would be $name + _spec. the _ is needed in the MagicDom to
+///     // extract the $name.
+///     struct DOMPoint_spec {
+///         x: f64,
+///         y: f64,
+///         z: f64,
+///         w: f64,
+///     }
+/// }
```

Comment:
Thanks for filling out the documentation!

Small nitpick: We wrap code examples in <code>```</code> so that they get rendered in a `<pre><code>...</code></pre>` by rustdoc and also compiled + tested by `cargo test`. If you want to disable the latter, you can do

    ```ignore
    example
    ```